frontend/store/vaults.ts

```diff
@@ -0,0 +1,74 @@
+import { Vault, VaultTransaction } from 'auctions-core/src/types';
+import Vue from 'vue';
+import { fetchVault, fetchVaultsCount, liquidateVault } from 'auctions-core/src/vaults';
+import { ActionContext } from 'vuex';
+
+interface State {
+    vaultTransactions: Record<Vault['id'], VaultTransaction>;
+    vaultsCount?: number;
+    isVaultLoading: boolean;
+}
+
+const getInitialState = (): State => ({
+    vaultTransactions: {},
+    vaultsCount: undefined,
+    isVaultLoading: false,
+});
+
+export const state = (): State => getInitialState();
+
+export const getters = {
+    vaultTransactions(state: State) {
+        return state.vaultTransactions;
+    },
+};
+export const mutations = {
+    setVault(state: State, vaultTransaction: VaultTransaction) {
+        Vue.set(state, 'vaults', vaultTransaction);
+    },
+    setIsVaultLoading(state: State, isLoading: boolean) {
+        state.isVaultLoading = isLoading;
+    },
+    setVaultsCount(state: State, count: number) {
+        state.vaultsCount = count;
+    },
+};
+export const actions = {
+    async fetchVault({ commit, rootGetters }: ActionContext<State, State>, vaultId: number) {
+        const network = rootGetters['network/getMakerNetwork'];
+        commit('setIsVaultLoading', true);
+        try {
+            const vaultTransaction = await fetchVault(network, vaultId);
```

Comment:
What happens here if I try to fetch index that doesn't exist?

Reply:
The error is thrown due to changes from the previous comments.